pkg/auth/authorizer/abac/abac.go

```diff
@@ -100,13 +101,20 @@ func NewFromFile(path string) (policyList, error) {
 func (p policy) matches(a authorizer.Attributes) bool {
 	if p.subjectMatches(a) {
 		if p.Readonly == false || (p.Readonly == a.IsReadOnly()) {
-			if p.Resource == "" || (p.Resource == a.GetResource()) {
+			switch {
+			case p.NonResourcePath != "":
```

Comment:
I'd like to preserve the information from `GetRequestInfo` in the attributes. If we did, this structure could look something like this:
```
// Ensure resource rules only match resource requests, and non-resource rules only match non-resource requests
isResourceRule := p.NonResourcePath == ""
if isResourceRule != a.IsResourceRequest() {
  return false
}

if isResourceRule {
  // Require namespace to match, if present in the policy rule
  if p.Namespace != "" && p.Namespace != a.GetNamespace() {
    return false
  }
  // Require resource to match, if present in the policy rule
  if p.Resource != "" && p.Resource != a.GetResource() {
    return false
  }
  return true
}

// Exact non-resource matches
if p.NonResourcePath == a.GetPath() {
  return true
}
// Subpath matches for `.../*` non-resource rules
if strings.HasSuffix(p.NonResourcePath, "/*") && strings.HasPrefix(a.GetPath(), string.TrimRight(p.NonResourcePath, "*")) {
  return true
}
return false
```